packages/composite-checkout/src/components/checkout-payment-methods.tsx

```diff
@@ -1,8 +1,11 @@
+import { useShoppingCart } from '@automattic/shopping-cart';
 import styled from '@emotion/styled';
 import { sprintf } from '@wordpress/i18n';
 import { useI18n } from '@wordpress/react-i18n';
 import debugFactory from 'debug';
 import { useCallback, useContext } from 'react';
+import { hasFreeCouponTransfersOnly } from 'calypso/lib/cart-values/cart-items';
+import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
```

Comment:
@Automattic/shilling Getting a "no restricted imports" lint error for these and I'm assuming you might be able to suggest a fix / other direction to go?

We did the same thing in https://github.com/Automattic/wp-calypso/pull/79920 which is also part of the composite checkout but there was no similar issue there.

Reply:
The difference from the other PR is that that file was from `client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.tsx` and here we're using `packages/composite-checkout/src/components/checkout-payment-methods.tsx`. So I guess there may be different lint rules set up for this?

Reply:
I guess a question unrelated to this PR would be, why are there two composite-checkout components 🤔

Reply:
It looks like the issue comes from this eslint config:

https://github.com/Automattic/wp-calypso/blob/update/checkout-copy/.eslintrc.js#L55

So, my guess is that we'll have to move whatever logic we need to a package.

Reply:
What @ebinnion said. 

The extended explanation for the warning is that we're trying to import something from [calypso's client](https://github.com/Automattic/wp-calypso/blob/update/checkout-copy/client/package.json) inside a package (`@automattic/composite-checkout`). That should generally not happen for a few reasons, but most specifically because the `@automattic/composite-checkout` package [is a dependency](https://github.com/Automattic/wp-calypso/blob/update/checkout-copy/client/package.json#L29) of Calypso's client, and as such, it doesn't make sense for the package to import from the `client`, as [circular dependencies](https://en.wikipedia.org/wiki/Circular_dependency) aren't recommended. 

As Eric suggested, the solution is to move the functionality to a package, and then import the functionality from the package. Looking at the functionality in question, there might be a few suitable packages for it, like `@automattic/calypso-products` or `@automattic/composite-checkout` or `@automattic/shopping-cart`.

Here are some additional materials to read about our Calypso monorepo setup and preferring to publish code in packages, in case those are helpful:

* p4TIVU-9vf-p2
* p4TIVU-arC-p2
* [monorepo docs](https://github.com/Automattic/wp-calypso/blob/trunk/docs/monorepo.md)

Let me know if that helps, and feel free to ping me or anyone else from @Automattic/team-calypso-frameworks if you have any questions.